From 50034ca2600ae07203f397af6d78e2e9ff16bc02 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Sat, 6 Oct 2018 18:12:23 -0700
Subject: [PATCH] Fix compiler warnings seen with musl

Fixes

checks.c:975:16: error: format '%ld' expects argument of type 'long
int', but argument 6 has type 'unsigned int' [-Werror=format=]
   FAIL(c, dti, "property '%s' size (%d) is invalid, expected multiple
of %ld in node %s",
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        prop->name, prop->val.len, sizeof(cell_t), node->fullpath);
                                   ~~~~~~~~~~~~~~
checks.c:96:23: note: in definition of macro 'FAIL'
   check_msg((c), dti, __VA_ARGS__);   \
                       ^~~~~~~~~~~
checks.c: In function 'check_interrupts_property':
checks.c:1166:16: error: format '%ld' expects argument of type 'long
int', but argument 6 has type 'unsigned int' [-Werror=format=]
   FAIL(c, dti, "property '%s' size (%d) is invalid, expected multiple
of %ld in node %s",
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        irq_prop->name, irq_prop->val.len, sizeof(cell_t),
                                           ~~~~~~~~~~~~~~

Upstream-Status: Backport [adapted from an upstream fix]

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 checks.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/checks.c b/checks.c
index 902f2e3..08a3a29 100644
--- a/checks.c
+++ b/checks.c
@@ -972,7 +972,7 @@ static void check_property_phandle_args(struct check *c,
 	int cell, cellsize = 0;
 
 	if (prop->val.len % sizeof(cell_t)) {
-		FAIL(c, dti, "property '%s' size (%d) is invalid, expected multiple of %ld in node %s",
+		FAIL(c, dti, "property '%s' size (%d) is invalid, expected multiple of %zu in node %s",
 		     prop->name, prop->val.len, sizeof(cell_t), node->fullpath);
 		return;
 	}
@@ -1163,7 +1163,7 @@ static void check_interrupts_property(struct check *c,
 		return;
 
 	if (irq_prop->val.len % sizeof(cell_t))
-		FAIL(c, dti, "property '%s' size (%d) is invalid, expected multiple of %ld in node %s",
+		FAIL(c, dti, "property '%s' size (%d) is invalid, expected multiple of %zu in node %s",
 		     irq_prop->name, irq_prop->val.len, sizeof(cell_t),
 		     node->fullpath);
 
